lib/new_relic/agent/instrumentation/middleware_tracing.rb

```diff
@@ -39,7 +39,7 @@ def build_transaction_options(env, first_middleware)
         end
 
         def merge_first_middleware_options(opts, env)
-          opts[:apdex_start_time] = QueueTime.parse_frontend_timestamp(env)
+          opts[:apdex_start_time] = QueueTime.frontend_timestamp(env)
```

Comment:
Changing the value we record for the apdex_start_time will have the undesired effect of changing the `apdex` scores that get computed.  `apdex` needs to retain visibility of the entire time the user is waiting for page updates.  Let's see if we can find a way to subtract the `puma.request_body_wait`

My first thought here is that we should record a metric for the `puma.request_body_wait` OR record the metric of the value computed here, and then a graph can render this data point in the UI.

Reply:
Thanks @mwlang, that makes a lot of sense.  I'm curious, just for clarification, it looks like the documentation refers to the`apdex` using response time as the metric.  https://docs.newrelic.com/docs/apm/new-relic-apm/apdex/apdex-measure-user-satisfaction
I guess my personal definition of response time wouldn't include the time it takes to upload a request body to the server.  Is it just that response time does normally upload time?  Or in a philosophical sense, we should include this time because this large payload for example is coming from our own front end app and we should know what the users are truly experiencing in terms of total time spent for the request?  Perhaps ideally a better name could be request time or something?

I like the idea of recording a metric for `puma.request_body_wait`.  Are you saying we would create a user-friendly name as a metric and populated it with `puma.request_body_wait` for Puma?  If we have a metric, it would be cool if we could just de-select it in the homepage chart if it's present or maybe just simply subtract it when using a query with the New Relic query language.